component/componenttest/assert_no_error_host.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package componenttest
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+// assertNoErrorHost implements a component.Host that asserts that
+// there were no errors.
+type assertNoErrorHost struct {
```

Comment:
For the moment please copy this code (do duplicate) where you need it, I really try to limit the public API we expose since we have to maintain this forever :)

Reply:
Done https://github.com/open-telemetry/opentelemetry-collector/pull/2679/commits/c41e58539f787e12891a6c5a0ae118c421c0d95b